### PR TITLE
Full update of the CiviCRM WP-CLI command

### PR DESCRIPTION
**Pull request by christianwach**
Overview
----------------------------------------
The `wp civicrm` command has not had any TLC for a long time. This update replaces it with a far more capable set of sub-commands as developed in the [Command Line Tools for CiviCRM](https://github.com/christianwach/cli-tools-for-civicrm#command-line-tools-for-civicrm) plugin.

Technical Details
----------------------------------------
My eventual aim is to have a separate repo (e.g. `civicrm/wp-cli-civicrm` or similar) created where the code in the current `wp-cli` directory can reside. This would be appropriately built so that it can be registered as a package that can be installed directly into WP-CLI itself. I'm going to need help setting that up, though. This PR at least makes the code that will form the basis of the new repo available here.

(FYI: a WP-CLI package can be installed with a command such as `wp package install git@github.com:civicrm/wp-cli-civicrm.git` or simply `wp package install civicrm/wp-cli-civicrm` if it's available via [Packagist](https://packagist.org/explore/?type=wp-cli-package&tags=wp%20cli) I think. I'm kinda new to the `composer` world so haven't worked out all the details yet. I've got as far as building a skeleton repo locally with `wp scaffold package` 🤷)

In terms of testing this PR, please refer to the readme file in the PR for an overview of the commands themselves. And, better still, use `wp help civicrm <command>`  for full details and usage of each command 😄 

Of particular note - there's a contradiction with the `wp civicrm core install` command in that it can't be called until CiviCRM is installed, at which point it is kinda redundant. It is included in preparation for building the package repo and can be tested by installing the [Command Line Tools for CiviCRM](https://github.com/christianwach/cli-tools-for-civicrm#command-line-tools-for-civicrm) plugin.

Comments
----------------------------------------
With many thanks to @JoeMurray for promising funding for this PR 🤩

### CODE REVIEW

**Comment by kcristiano** [comment]
same test failure that seems unrelated.....

**Comment by kcristiano** [comment]
@christianwach Thank you for this.   I've built a tarball using distmaker and will add to some sites to test.

preliminary testing looks great

File: wp-cli/README.md [hunk 1]
```diff
@@ -1,88 +1,102 @@
...
+| `wp civicrm disable-debug` | `wp civicrm debug disable` |
+| `wp civicrm enable-debug` | `wp civicrm debug enable` |
+| `wp civicrm install` | See [Composite Commands](#composite-commands) below |
+| `wp civicrm member-records` | `wp civicrm job member-records` or `wp civicrm job membership` |
+| `wp civicrm process-mail-queue` | `wp civicrm job process-mail-queue` or `wp civicrm job mailing` |
+| `wp civicrm restore` | Only works with backups made by `wp civicrm upgrade` |
+| `wp civicrm sql-conf` | `wp civicrm db config` or `wp civicrm db conf` |
+| `wp civicrm sql-connect` | `wp civicrm db connect` |
+| `wp civicrm sql-cli` | `wp civicrm db cli` |
+| `wp civicrm sql-dump` | `wp civicrm db export` |
```
Comment (kcristiano) [inline wp-cli/README.md:74]: I'd remove this from 'deprecated' as these command do very different things.   `sql-dump` is the full DB and `db export is just the known CiviCRM tables

Comment (christianwach) [reply wp-cli/README.md]: @kcristiano Good point. However, in the interests of cleaning up the top level of sub-commands, I've kept `wp civicrm sql-dump` as deprecated, un-aliased `wp civicrm db dump` and moved the functionality there in ca4a000d73618560ea6c12f1f9344e0c680752d5

Comment (christianwach) [reply wp-cli/README.md]: FYI the subcommands should eventually become:

```
SYNOPSIS

  wp civicrm <command>

SUBCOMMANDS

  api                     Access the CiviCRM API v3.
  api3                    Access the CiviCRM API v3.
  cache                   Flush the CiviCRM cache.
  core                    Downloads, installs, updates, and manages a CiviCRM installation.
  db                      Utilities for interacting with the CiviCRM database.
  debug                   Enable or disable debugging in CiviCRM.
  job                     Run CiviCRM cron jobs.
  pipe                    Start a Civi::pipe session (JSON-RPC 2.0)
```

There will be `api4` and `ext` coming next, plus any others that people write. I hope you agree that this list is more logical than the previous scattergun approach to the naming scheme of subcommands.

**Comment by kcristiano** [comment]
Test fail is known and unrelated.

This PR should be in as early as possible for testing.   Merging

**Pull request merged**